src/DssDirectDepositHub.sol

```diff
@@ -0,0 +1,571 @@
+        // This module will have an unintended behaviour if rate is changed to some other value.
+        if (amount == 0) {
+            emit Wind(ilk, 0);
+            return;
+        }
+
+        require(int256(amount) >= 0, "DssDirectDepositHub/overflow");
+
+        vat.slip(ilk, address(pool), int256(amount));
+        vat.frob(ilk, address(pool), address(pool), address(this), int256(amount), int256(amount));
+        // normalized debt == erc20 DAI (Vat rate for this ilk fixed to 1 RAY)
+        daiJoin.exit(address(pool), amount);
+        pool.deposit(amount);
+
+        emit Wind(ilk, amount);
+    }
+
+    function _unwind(bytes32 ilk, D3MPoolLike pool, uint256 supplyReduction, uint256 availableAssets, Mode mode, uint256 assetBalance) internal {
+        // IMPORTANT: this function assumes Vat rate of this ilk will always be == 1 * RAY (no fees).
+        // That's why it converts normalized debt (art) to Vat DAI generated with a simple RAY multiplication or division
+        // This module will have an unintended behaviour if rate is changed to some other value.
+
+        EndLike end_;
+        uint256 daiDebt;
+        if (mode == Mode.NORMAL) {
+            // Normal mode or module just caged (no culled)
+            // debt is obtained from CDP art
+            (,daiDebt) = vat.urns(ilk, address(pool));
+        } else if (mode == Mode.MODULE_CULLED) {
+            // Module shutdown and culled
+            // debt is obtained from free collateral owned by this contract
+            // We rebalance the CDP after grabbing in `cull` so the gems represents
+            // the debt at time of cull
+            daiDebt = vat.gem(ilk, address(pool));
+        } else {
+            // MCD caged
+            // debt is obtained from free collateral owned by the End module
+            end_ = end;
+            end_.skim(ilk, address(pool));
+            daiDebt = vat.gem(ilk, address(end_));
+        }
+
+        // Unwind amount is limited by how much:
+        // - max reduction desired
+        // - assets available
+        // - dai debt tracked in vat (CDP or free)
+        uint256 amount = _min(
+                            _min(
+                                supplyReduction,
+                                availableAssets
+                            ),
+                            daiDebt
+                        );
+
+        // Determine the amount of fees to bring back
+        uint256 fees = 0;
+        if (assetBalance > daiDebt) {
+            fees = assetBalance - daiDebt;
+
+            if (_add(amount, fees) > availableAssets) {
+                // Don't need safe-math because this is constrained above
+                fees = availableAssets - amount;
+            }
+        }
+
+        if (amount == 0 && fees == 0) {
+            emit Unwind(ilk, 0, 0);
+            return;
+        }
+
+        require(amount <= 2 ** 255, "DssDirectDepositHub/overflow");
```

Comment:
@gbalabasquer suggestion to move 2 ** 255 to min setting in amount:

```suggestion
        uint256 amount = _min(
                                         _min(
                                            _min(
                                                 supplyReduction,
                                                 availableAssets
                                            ),
                                           daiDebt
                                         ), 
                                         2 ** 255
                                       );

        // Determine the amount of fees to bring back
        uint256 fees = 0;
        if (assetBalance > daiDebt) {
            fees = assetBalance - daiDebt;

            if (_add(amount, fees) > availableAssets) {
                // Don't need safe-math because this is constrained above
                fees = availableAssets - amount;
            }
        }

        if (amount == 0 && fees == 0) {
            emit Unwind(ilk, 0, 0);
            return;
        }
```